.travis.yml

```diff
@@ -0,0 +1,73 @@
+# Copyright 2015 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+sudo: false
+language: ruby
+rvm:
+- 2.2.3
+
+cache:
+  - directories:
+    - $HOME/gcloud/
+
+env:
+  global:
+  - PATH=$PATH:$HOME/gcloud/google-cloud-sdk/bin
+  - GOOGLE_APPLICATION_CREDENTIALS=$TRAVIS_BUILD_DIR/client_secrets.json
+  matrix:
+  - STEP_NAME=1-hello-world
+  # - STEP_NAME=2-cloud-datastore
+  # - STEP_NAME=2-cloud-sql
+  # - STEP_NAME=2-postgresql
+  # - STEP_NAME=3-cloud-storage
+  # - STEP_NAME=4-auth
+  # - STEP_NAME=5-logging
+  # - STEP_NAME=6-task-queueing
+  # - STEP_NAME=7-compute-engine
+
+before_install:
+#ENCRYPT YOUR PRIVATE KEY (If you need authentication)
+# 1. Install and login to the Travis CLI:
+#       $ gem install travis
+#       $ travis login
+# 2. Move your json private key to client_secrets.json
+# 3. Run:
+#       $ travis encrypt-file client_secrets.json --add
+# 4. Commit changes:
+#       $ git add client_secrets.json.enc
+#       $ git commit client_secrets.json.enc .travis.yml
+- openssl aes-256-cbc -K $encrypted_c66238cfcad7_key -iv $encrypted_c66238cfcad7_iv
+  -in client_secrets.json.enc -out client_secrets.json -d
+- if [ ! -d $HOME/gcloud/google-cloud-sdk ]; then
+  mkdir -p $HOME/gcloud &&
+  wget https://dl.google.com/dl/cloudsdk/release/google-cloud-sdk.tar.gz --directory-prefix=$HOME/gcloud &&
+  cd $HOME/gcloud &&
+  tar xzf google-cloud-sdk.tar.gz &&
+  printf '\ny\n\ny\ny\n' | ./google-cloud-sdk/install.sh &&
+  cd $TRAVIS_BUILD_DIR;
+  fi
+- gcloud -q components update
+- if [ -a client_secrets.json ]; then
+  gcloud -q auth activate-service-account --key-file client_secrets.json;
+  fi
+
+install:
+- bundle install --jobs=3 --retry=3
+
+script:
+- bundle exec rspec spec
```

Comment:
We're `bundle exec` running commands but there is no `Gemfile` to define the bundle.

Might I suggest something like ...

```
spec/Gemfile
spec/Gemfile.lock       # created by running `bundle install` and should be checked into repo
spec/e2e.rb             # fwiw we could make e2e an RSpec test by itself, but script-style works too
spec/e2e_cleanup.rb
spec/steps/1_hello_world_spec.rb
```

And Travis could `cd spec && bundle install && bundle exec rspec e2e.rb`

Although it might be cleaner to simply have ...

```
spec/Gemfile
spec/Gemfile.lock
spec/spec_helper.rb          # this can contain a method like `def deploy`
spec/1_hello_world_spec.rb   # has a `before :all` that deploys ... each `it` block tests functionality ... `after :all` tears down
```